src/components/FiltersModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import Modal from "react-modal";
+
+Modal.setAppElement("#root");
+
+type TFiltersModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  localFilters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  };
+  setLocalFilters: (filters: {
+    isOnView: boolean;
+    isPublicDomain: boolean;
+    isHiddenGem: boolean;
+    searchQuery: string;
+  }) => void;
+  onSubmit: (e: React.FormEvent) => void;
+};
+
+const FiltersModal: React.FC<TFiltersModalProps> = ({
+  isOpen,
+  onClose,
+  localFilters,
+  setLocalFilters,
+  onSubmit,
+}) => {
+  const toggleFilter = (key: keyof typeof localFilters) => {
+    setLocalFilters({
+      ...localFilters,
+      [key]: !localFilters[key],
+    });
```

Comment:
The generic approach looks good, however, this should be an internal state of the `FiltersModal` component